src/pages/Profile.js

```diff
@@ -58,6 +60,120 @@ const styles = {
 }
 
 
+const NihLink = ({ nihToken }) => {
+  /*
+   * Hooks
+   */
+  const [{ linkedNihUsername, linkExpireTime, datasetPermissions }, setNihStatus] = useState({})
+  const [loading, setLoading] = useState(false)
+  const [linking, setLinking] = useState(false)
+  const signal = useCancellation()
+
+  Utils.useOnMount(() => {
+    const { User } = Ajax(signal)
+
+    const linkNihAccount = _.flow(
+      withErrorReporting('Error linking NIH account'),
+      Utils.withBusyState(setLinking)
+    )(async nihToken => {
+      setNihStatus(await User.linkNihAccount(nihToken))
+    })
+
+    const loadNihStatus = _.flow(
+      withErrorReporting('Error loading NIH account status'),
+      Utils.withBusyState(setLoading)
+    )(async () => {
+      try {
+        setNihStatus(await User.getNihStatus())
+      } catch (error) {
+        if (error.status === 404) setNihStatus({})
+        else throw error
+      }
+    })
+
+    if (nihToken) {
+      // Clear the query string, but use replace so the back button doesn't take the user back to the token
+      Nav.history.replace({ search: '' })
+      linkNihAccount(nihToken)
```

Comment:
Possibly. There is some trickiness around doing the redirect regardless of outcome, reporting the error if there is one, and capturing the result of the link call which avoids calling the status endpoint immediately after linking. That last point probably isn't a huge deal, but I'm looking at some refactoring to put the NIH link status in `authStore` so that the global link check and the rendering on the profile page can work from the same data. It might be easier to act on this suggestion after that is done. Does that work for you @panentheos?

Reply:
Sounds good!